### PR TITLE
feat: add no_whitespace parameter

### LINKED ISSUE #4: [feat] - parameter to handle trail and starting whitespaces
Create a parameter, maybe `remove_trail` or `no_whitespace` , to remove before the parameter creation the whitespaces

### PR DESCRIPTION
**Pull request by ilyichv**
## Description

Add a new parameter `no_whitespace` for trimming an removing whitespaces automatically from parameters. 
By default, it's now applied to `id` and `arke_id`

## Related Issue

Resolve #4 

@arkemishub/backend-experts 
Let me know if you prefer to add it somewhere else, I set it to false for most of default Units

### CODE REVIEW

**Review by ErikFerrari — APPROVED** [review]

**Comment by ilyichv** [comment]
@ErikFerrari proposed to rename `no_whitespace` into `strip`. I actually like it, what do you think @dorianmercatante?

File: lib/arke.ex [hunk 1]
```diff
@@ -101,7 +101,8 @@ defmodule Arke do
             values: nil,
             multiple: false,
             unique: true,
-            default_string: nil
+            default_string: nil,
+            no_whitespace: true
```
Comment (ErikFerrari) [inline lib/arke.ex:105]:
```suggestion
            strip: true
```

File: lib/arke.ex [hunk 2]
```diff
@@ -128,7 +129,8 @@ defmodule Arke do
             values: nil,
             multiple: false,
             unique: false,
-            default_string: nil
+            default_string: nil,
+            no_whitespace: true
```
Comment (ErikFerrari) [inline lib/arke.ex:133]:
```suggestion
            strip: true
```

File: lib/arke.ex [hunk 3]
```diff
@@ -222,6 +224,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: 200,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:227]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 4]
```diff
@@ -244,6 +247,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:250]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 5]
```diff
@@ -311,6 +315,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:318]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 6]
```diff
@@ -336,6 +341,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:344]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 7]
```diff
@@ -380,6 +386,21 @@ defmodule Arke do
...
+          base_parameter(label: "No Whitespace"),
+          %{default_boolean: false}
+        ),
+        :boolean,
+        nil,
+        %{},
+        nil,
+        nil,
+        nil
+      )
```
Comment (ErikFerrari) [inline lib/arke.ex:402]:
```suggestion
    strip =
      Unit.new(
        :strip,
        Map.merge(
          base_parameter(label: "Strip Whitespace"),
          %{default_boolean: false}
        ),
        :boolean,
        nil,
        %{},
        nil,
        nil,
        nil
      )
```

File: lib/arke.ex [hunk 8]
```diff
@@ -493,6 +514,7 @@ defmodule Arke do
           %{
             min_length: nil,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:517]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 9]
```diff
@@ -657,6 +679,7 @@ defmodule Arke do
           %{
             min_length: 0,
             max_length: 500,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:682]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 10]
```diff
@@ -709,6 +732,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: nil,
+            no_whitespace: true,
```
Comment (ErikFerrari) [inline lib/arke.ex:735]:
```suggestion
            strip: true,
```

File: lib/arke.ex [hunk 11]
```diff
@@ -731,6 +755,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:758]:
```suggestion
            strip: true,
```

File: lib/arke.ex [hunk 12]
```diff
@@ -753,6 +778,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:781]:
```suggestion
            strip: true,
```

File: lib/arke.ex [hunk 13]
```diff
@@ -775,6 +801,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: 100,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:804]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 14]
```diff
@@ -797,6 +824,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:827]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 15]
```diff
@@ -819,6 +847,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: nil,
+            no_whitespace: true,
```
Comment (ErikFerrari) [inline lib/arke.ex:850]:
```suggestion
            strip: true,
```

File: lib/arke.ex [hunk 16]
```diff
@@ -841,6 +870,7 @@ defmodule Arke do
           %{
             min_length: 2,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:873]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 17]
```diff
@@ -863,6 +893,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:896]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 18]
```diff
@@ -885,6 +916,7 @@ defmodule Arke do
           %{
             min_length: 3,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:919]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 19]
```diff
@@ -907,6 +939,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:942]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 20]
```diff
@@ -929,6 +962,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:965]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 21]
```diff
@@ -951,6 +985,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:988]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 22]
```diff
@@ -973,6 +1008,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:1011]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 23]
```diff
@@ -1010,6 +1046,7 @@ defmodule Arke do
           %{
             min_length: 1,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:1049]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 24]
```diff
@@ -1032,6 +1069,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:1072]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 25]
```diff
@@ -1084,6 +1122,7 @@ defmodule Arke do
           %{
             min_length: 1,
             max_length: nil,
+            no_whitespace: true,
```
Comment (ErikFerrari) [inline lib/arke.ex:1125]:
```suggestion
            strip: true,
```

File: lib/arke.ex [hunk 26]
```diff
@@ -1140,6 +1179,7 @@ defmodule Arke do
           %{
             min_length: nil,
             max_length: nil,
+            no_whitespace: false,
```
Comment (ErikFerrari) [inline lib/arke.ex:1182]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 27]
```diff
@@ -1177,6 +1217,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:1220]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 28]
```diff
@@ -1199,6 +1240,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:1243]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 29]
```diff
@@ -1221,6 +1263,7 @@ defmodule Arke do
...
```
Comment (ErikFerrari) [inline lib/arke.ex:1266]:
```suggestion
            strip: false,
```

File: lib/arke.ex [hunk 30]
```diff
@@ -1280,6 +1323,7 @@ defmodule Arke do
       required,
       persistence,
       helper_text,
+      no_whitespace,
```
Comment (ErikFerrari) [inline lib/arke.ex:1326]:
```suggestion
      strip,
```

File: lib/arke/core/parameter.ex [hunk 31]
```diff
@@ -126,6 +126,7 @@ defmodule Arke.Core.Parameter.String do
     Parameter.base_parameters()
     parameter(:min_length, :integer, required: false)
     parameter(:max_length, :integer, required: false)
+    parameter(:no_whitespace, :boolean, default_boolean: false)
```
Comment (ErikFerrari) [inline lib/arke/core/parameter.ex:129]:
```suggestion
    parameter(:strip, :boolean, default_boolean: false)
```

File: lib/arke/struct_manager.ex [hunk 32]
```diff
@@ -386,6 +386,7 @@ defmodule Arke.StructManager do
     Map.merge(base_data, %{
       max_length: data.max_length,
       min_length: data.min_length,
+      no_whitespace: data.no_whitespace,
```
Comment (ErikFerrari) [inline lib/arke/struct_manager.ex:389]:
```suggestion
      strip: data.strip,
```

File: lib/arke/system.ex [hunk 33]
```diff
@@ -283,6 +283,7 @@ defmodule Arke.System.BaseParameter do
     |> parameter_option(:type, :string)
     |> parameter_option(:min_length, nil)
     |> parameter_option(:max_length, nil)
+    |> parameter_option(:no_whitespace, nil)
```
Comment (ErikFerrari) [inline lib/arke/system.ex:286]:
```suggestion
    |> parameter_option(:strip, nil)
```

File: lib/arke/validator.ex [hunk 34]
```diff
@@ -294,6 +295,13 @@ defmodule Arke.Validator do
     Enum.all?(value, &condition.(&1))
   end
 
+  defp check_no_whitespace(%{data: %{no_whitespace: true}} = parameter, value) do
```
Comment (ErikFerrari) [inline lib/arke/validator.ex:298]:
```suggestion
  defp check_no_whitespace(%{data: %{strip: true}} = parameter, value) do
```

**Review by ErikFerrari — COMMENTED** [review]
Changed all the `no_whitespace` to `strip`

File: lib/arke.ex [hunk 35]
```diff
@@ -379,6 +385,21 @@ defmodule Arke do
         nil
       )
 
+    strip =
+      Unit.new(
+        :strip,
+        Map.merge(
+          base_parameter(label: "No Whitespace"),
```
Comment (ErikFerrari) [inline lib/arke.ex:392]:
```suggestion
          base_parameter(label: "Remove Whitespace"),
```

**Review by ErikFerrari — APPROVED** [review]

**Pull request merged**